src/odbc/odbc.c

```diff
@@ -1453,54 +1513,105 @@ _SQLBindParameter(SQLHSTMT hstmt, SQLUSMALLINT ipar, SQLSMALLINT fParamType, SQL
-	}
-	drec = &apd->records[ipar - 1];
+	/* If the parameter focus is set to 0, bind values as arguments */
+	/* Otherwise, bind values as the columns of a table-valued parameter */
+	if (stmt->param_focus == 0) {
+		/* fill APD related fields */
+		apd = stmt->apd;
+		orig_apd_size = apd->header.sql_desc_count;
+		if (ipar > apd->header.sql_desc_count && desc_alloc_records(apd, ipar) != SQL_SUCCESS) {
+			odbc_errs_add(&stmt->errs, "HY001", NULL);
+			ODBC_EXIT_(stmt);
+		}
+		drec = &apd->records[ipar - 1];
 
-	if (odbc_set_concise_c_type(fCType, drec, 0) != SQL_SUCCESS) {
-		desc_alloc_records(apd, orig_apd_size);
-		odbc_errs_add(&stmt->errs, "HY004", NULL);
-		ODBC_EXIT_(stmt);
-	}
+		if (odbc_set_concise_c_type(fCType, drec, 0) != SQL_SUCCESS) {
+			desc_alloc_records(apd, orig_apd_size);
+			odbc_errs_add(&stmt->errs, "HY004", NULL);
+			ODBC_EXIT_(stmt);
+		}
 
-	stmt->need_reprepare = 1;
+		stmt->need_reprepare = 1;
 
-	/* TODO other types ?? handle SQL_C_DEFAULT */
-	if (drec->sql_desc_type == SQL_C_CHAR || drec->sql_desc_type == SQL_C_WCHAR || drec->sql_desc_type == SQL_C_BINARY)
-		drec->sql_desc_octet_length = cbValueMax;
-	drec->sql_desc_indicator_ptr = pcbValue;
-	drec->sql_desc_octet_length_ptr = pcbValue;
-	drec->sql_desc_data_ptr = (char *) rgbValue;
+		/* TODO other types ?? handle SQL_C_DEFAULT */
+		if (drec->sql_desc_type == SQL_C_CHAR || drec->sql_desc_type == SQL_C_WCHAR || drec->sql_desc_type == SQL_C_BINARY)
+			drec->sql_desc_octet_length = cbValueMax;
+		drec->sql_desc_indicator_ptr = pcbValue;
+		drec->sql_desc_octet_length_ptr = pcbValue;
+		drec->sql_desc_data_ptr = (char *) rgbValue;
 
-	/* field IPD related fields */
-	ipd = stmt->ipd;
-	orig_ipd_size = ipd->header.sql_desc_count;
-	if (ipar > ipd->header.sql_desc_count && desc_alloc_records(ipd, ipar) != SQL_SUCCESS) {
-		desc_alloc_records(apd, orig_apd_size);
-		odbc_errs_add(&stmt->errs, "HY001", NULL);
-		ODBC_EXIT_(stmt);
-	}
-	drec = &ipd->records[ipar - 1];
+		if (fSqlType == SQL_SS_TABLE) {
+			/* For a table value, the pcbValue is a buffer containing the number of rows present */
+			/* Replace it with a buffer to the acutal size of a TDS_TVP struct */
+			/* TODO: Replace with a factory method or global variable? */
+			drec->sql_desc_octet_length_ptr = malloc(sizeof(SQLLEN));
+			*(drec->sql_desc_octet_length_ptr) = sizeof(SQLTVP);
 
-	drec->sql_desc_parameter_type = fParamType;
-	if (odbc_set_concise_sql_type(fSqlType, drec, 0) != SQL_SUCCESS) {
-		desc_alloc_records(ipd, orig_ipd_size);
-		desc_alloc_records(apd, orig_apd_size);
-		odbc_errs_add(&stmt->errs, "HY004", NULL);
-		ODBC_EXIT_(stmt);
-	}
-	if (is_numeric) {
-		drec->sql_desc_precision = cbColDef;
-		drec->sql_desc_scale = ibScale;
+			tvp_name_len = cbValueMax == SQL_NTS ? strlen(rgbValue) : cbValueMax;
+			if ((tvp_name = malloc(tvp_name_len + 1)) == NULL) {
+				odbc_errs_add(&stmt->errs, "HY001", NULL);
+				ODBC_EXIT_(stmt);
+			}
+
+			strncpy(tvp_name, rgbValue, tvp_name_len);
```

Comment:
Probably just `memcpy` the `tvp_name_len + 1` size here.

Reply:
Out of curiosity, are `strncpy` and `memcpy` compiled differently?

For this, I'm adding in the null terminator because the char buffer may not be null-terminated, and the length is specified via `cbValueMax` (line 1550).